File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        profit = 0 
        L = 0 
        R = 1 
        
        while R <= len(prices) - 1: 
            currProfit = prices[R] - prices[L]
            profit = max(currProfit, profit) 
            
            if prices[L] > prices[R]:
                L = R 
                
            R += 1 
        
        return profit 
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0; 
        
        int leftPointer = 0; 
        int rightPointer = 1; 
        
        while (rightPointer <= prices.length - 1) {
            int currentProfit = prices[rightPointer] - prices[leftPointer]; 
            profit = Integer.max(profit, currentProfit); 
            
            if (prices[leftPointer] > prices[rightPointer]) {
                leftPointer = rightPointer; 
            }
            
            rightPointer += 1; 
        }
        
        return profit; 
        
                
    }
}